openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TeamRepository.java

```diff
@@ -230,13 +253,19 @@ public void storeRelationships(Team team) {
   public void setInheritedFields(Team team, Fields fields) {
     // If user does not have domain, then inherit it from parent Team
     // TODO have default team when a user belongs to multiple teams
-    if (fields.contains(FIELD_DOMAIN) && team.getDomain() == null) {
+    if (fields.contains("teamDomains")) {
+      Set<EntityReference> combinedParent = new TreeSet<>(EntityUtil.compareEntityReferenceById);
       List<EntityReference> parents =
           !fields.contains(PARENTS_FIELD) ? getParents(team) : team.getParents();
       if (!nullOrEmpty(parents)) {
-        Team parent = Entity.getEntity(TEAM, parents.get(0).getId(), "domain", ALL);
-        inheritDomain(team, fields, parent);
+        for (EntityReference parentRef : parents) {
+          Team parentTeam = Entity.getEntity(TEAM, parentRef.getId(), "teamDomains", ALL);
+          combinedParent.addAll(parentTeam.getTeamDomains());
+        }
       }
+      team.setTeamDomains(
```

Comment:
why are we expecting multiple domains to be inherited from?

Reply:
Parent teams